### ISSUE TITLE
[Bug]: New free IP address from prefix is not assigned to the virtual machine interface

### ISSUE THREAD

**Issue by mlvgn** [netbox-community/ansible_modules#1075]
### Ansible NetBox Collection version

v3.14.0

### Ansible version

```bash
<!--- Paste verbatim output from "ansible --version" between quotes -->
ansible [core 2.14.4]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.11/site-packages/ansible
  ansible collection location = /home/user/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.11.3 (main, Apr  5 2023, 00:00:00) [GCC 12.2.1 20221121 (Red Hat 12.2.1-4)] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

### NetBox version

v3.6.3

### Python version

3.11

### Steps to Reproduce

1. Create a VM test in NetBox
2. Add an interface named eth0 to it
3. Create prefix 10.0.0.0/24
4. Run task

```
  - name: Add IP addresses to NetBox and assign to virtual machine
    netbox.netbox.netbox_ip_address:
      data:
        prefix: "10.0.0.0/24"
        dns_name: "test.example.org"
        assigned_object:
          name: "eth0"
          virtual_machine: "test"
      state: new
      # Connection settings
      netbox_url: "https://netbox.example.org"
      netbox_token: "..."
      validate_certs: '/etc/pki/tls/certs/ca-bundle.crt'
```

### Expected Behavior

An IP address was created but not assigned to the virtual machine interface:

```
changed: [localhost] => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python3"
    },
    "changed": true,
    "diff": {
        "after": {
            "state": "present"
        },
        "before": {
            "state": "absent"
        }
    },
    "failed_when_result": false,
    "invocation": {
        "module_args": {
            "cert": null,
            "data": {
                "address": null,
                "assigned_object": {
                    "device": null,
                    "name": "eth0",
                    "virtual_machine": "test"
                },
                "comments": null,
                "custom_fields": null,
                "description": null,
                "dns_name": "test.example.org",
                "family": null,
                "interface": null,
                "nat_inside": null,
                "prefix": "10.0.0.0/24",
                "role": null,
                "status": null,
                "tags": null,
                "tenant": null,
                "vrf": null
            },
            "netbox_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "netbox_url": "https://netbox.example.org",
            "query_params": null,
            "state": "new",
            "validate_certs": "/etc/pki/tls/certs/ca-bundle.crt"
        }
    },
    "ip_address": {
        "address": "10.0.0.1/24",
        "assigned_object": null,
        "assigned_object_id": null,
        "assigned_object_type": null,
        "comments": "",
        "created": "2023-09-27T11:41:28.185442Z",
        "custom_fields": {},
        "description": "",
        "display": "10.0.0.1/24",
        "dns_name": "",
        "family": 4,
        "id": 439,
        "last_updated": "2023-09-27T11:41:28.185458Z",
        "nat_inside": null,
        "nat_outside": [],
        "role": null,
        "status": "active",
        "tags": [],
        "tenant": null,
        "url": "https://netbox.example.org/api/ipam/ip-addresses/439/",
        "vrf": null
    },
    "msg": "ip_address 10.0.0.1/24 created"
}

PLAY RECAP ********************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0  
```

### Observed Behavior

An IP address has been created and assigned to eth0 for the test virtual machine. Previously, this playbook worked with NetBox 3.4

**Comment by rodvand** [comment]
https://github.com/netbox-community/netbox/issues/13876

**Comment by rodvand** [comment]
Fixed with NetBox release 3.6.4.
